### PR TITLE
Add support for django 4.0

### PR DESCRIPTION
**Pull request by everspader**
The original repository for django-mfa only supports
django versions up to 3.2.

There are many interesting security patches in the
subsequent versions that make the upgrade to 4.0 worth it.

This requires that django-mfa is also updated to replace
removed methods that are no longer available.

Since we are forking the original repository, I rebased
it on `master` branch of the latest release and added a
commit with the required fix

- [ ] Update the requirements and other package files
(nice to have but not required since this is not distributed)

### CODE REVIEW

**Review by marius-cristian — APPROVED** [review]

File: django_mfa/views.py [hunk 1]
```diff
@@ -14,8 +14,13 @@
 from django.contrib.auth import load_backend
 from django.contrib import auth, messages
 from django.urls import reverse, reverse_lazy
-from django.utils.http import is_safe_url
-from django.utils.translation import ugettext as _
+try:
+    from django.utils.http import url_has_allowed_host_and_scheme as is_safe_url
```
Comment (laumann) [inline django_mfa/views.py:18]: how do you know that `url_has_allowed_host_and_scheme` is the proper replacement?

File: django_mfa/views.py [hunk 2]
```diff
@@ -14,8 +14,13 @@
...
+except ImportError:
+    # Django < 4.0 fallback
+    from django.utils.http import is_safe_url
+
+from django.utils.translation import gettext as _
```
Comment (laumann) [inline django_mfa/views.py:23]: did you mean to change ugettext to gettext?

File: django_mfa/views.py (hunk 1 above)
Comment (everspader) [reply django_mfa/views.py:18]: Going through all the release notes. Last bullet point here:
https://docs.djangoproject.com/en/4.0/releases/3.0/#miscellaneous-1

File: django_mfa/views.py (hunk 2 above)
Comment (everspader) [reply django_mfa/views.py:23]: Yes. Second bullet point here: https://docs.djangoproject.com/en/4.0/releases/3.0/#miscellaneous-1

**Pull request merged**